common/src/main/java/com/wynntils/wynn/model/SplashManager.java

```diff
@@ -34,21 +31,16 @@ public static void init() {
     }
 
     private static void updateCurrentSplash() {
-        RequestHandler handler = WebManager.getHandler();
-        handler.addAndDispatch(new RequestBuilder(SPLASHES_URL, "splashes")
-                .cacheTo(new File(WebManager.API_CACHE_ROOT, "splashes.json"))
-                .useCacheAsBackup()
-                .handleJsonArray(json -> {
-                    Type type = new TypeToken<List<String>>() {}.getType();
-                    allSplashes = WynntilsMod.GSON.fromJson(json, type);
-                    if (allSplashes.isEmpty()) {
-                        // Use fallback in case of failure
-                        currentSplash = DEFAULT_SPLASH;
-                    } else {
-                        currentSplash = allSplashes.get(Utils.getRandom().nextInt(allSplashes.size()));
-                    }
-                    return true;
-                })
-                .build());
+        Download dl = NetManager.download(UrlId.DATA_STATIC_SPLASHES);
+        dl.handleReader(reader -> {
+            Type type = new TypeToken<List<String>>() {}.getType();
```

Comment:
I don't think such "simple" types cannot be handled by GSON without a Type, but this should do.

Reply:
What do you mean? That I should use the GSON.fromJson(reader, Class<?> class) version instead? It clearly says "It is not suitable to use if the specified class is a generic type since it will not have the generic type information because of the Type Erasure feature of Java. Therefore, this method should not be used if the desired type is a generic type. "

Reply:
I remember a GSON method without a Type/Class. Anyways, I think I miss remember. I've been debugging matrixes for an hour before writing this 🙃.

Reply:
No, there is not. But if you don't have generics, you can do like `GSON.fromJson(reader, FluffyProfile.class)`.